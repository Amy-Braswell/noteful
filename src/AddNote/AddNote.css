.AddNote__Form {
    padding: 0;
    float: left;
    text-align: left;
    width: 40%;
  }

.AddNote h3 {
    margin: 130px 0 0;
    padding: 1vh 15px;
    color: #bdbdbd;
}

.AddNote_form-group input{
    background-color: #030f1f;
    border: none;
    outline: none;
    color: #b5b3b3; 
    display: block;
    font-family: var(--custom-font);
    font-size: 20px;
    margin: 1vh 0 1vh auto;
    /* max-width: 180px; */
    padding: 1vh 0 1vh 16px;
}

.new__note__name {
    padding: 0 15px;
    width: 100%;
}

.new__note__content {
    padding: 0 15px;
    margin-top: 1vh;
    margin-bottom: 1vh;
    height: 35vh;
    width: 100%;
}

input:-webkit-autofill::first-line {
    font-size: 20px
}

input:-webkit-autofill, input:-webkit-autofill:hover, input:-webkit-autofill:focus,
input, input[type=text] {
    border: none;
    font-size: 20px;
    font-family: var(--custom-font);
    padding: 8px 16px;
    background-color: #030f1f;
    -webkit-box-shadow: 0 0 0px 1000px  #030f1f inset;
    -webkit-text-fill-color: #b5b3b3;
    color: #b5b3b3;
    box-shadow: 0 0 0px 1000px  #030f1f inset;
    outline: none;
}

.AddNote__add-note-button{
margin: 2vh auto;
}

select {
    display: block;
    height: 3em;
    width: 100%;
    font-size: 1em;
    background-color: #030f1f;
    color: #b5b3b3;
    border: none;
    border-radius: 0px;
    -webkit-border-radius: 0px;
    box-sizing: border-box;
  }