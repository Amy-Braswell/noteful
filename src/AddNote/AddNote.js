import React, {Component} from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import NotePageNav from '../NotePageNav/NotePageNav'
import CircleButton from '../CircleButton/CircleButton'
import ApiContext from '../ApiContext'
import config from '../config'

class AddNote extends Component {
    constructor(props) {
        super(props)
        this.state = {
            name: {
                value: ''
            },
            id: {
                value: ''
            },
            content: {
                value: ''
            },
            error: {
                value: null
            }
        }
    }

    static contextType = ApiContext

    updateName(name) {
        this.setState({name: {value: name}})
    }

    updateContent(content) {
        this.setState({content: {value: content}})
    }

    handleSubmit(event) {
        event.preventDefault()
        const modified = new Date()
        const newNote = {
            name: this.state.name.value,
            content: this.state.content.value,
            modified: modified
        }
        const specificEndpoint = `${config.API_ENDPOINT}/notes`
        
       
        this.setState({ error: null })
        fetch(specificEndpoint, {
            headers: {
                'Content-Type': 'application/json'
            },
            method: 'POST',
            body: JSON.stringify(newNote)
        })

        .then(res => {
            if(!res.ok) {
                //get the error message from the response
                return res.json().then(error => {
                    // then throw it
                    throw error
                })
            }
            return res.json()
        })

        .then(data => {
            this.context.addNote(data);
            this.props.history.push('/')
        })

        .catch(error => {
            this.setState({error})
        })
    }

    render(){
        return(
            <div className='parent-div'>
            <NotePageNav />
            <form className='form'>
                <h2>Add a Note</h2>
                <div className='form-group'>
                    <input 
                        type='text' 
                        className='new__note__name'
                        name='name' 
                        id='name' 
                        onChange={e => this.updateName(e.target.value)}
                    />  
                </div>
                <div className='form-group'>
                    <input 
                        type='text' 
                        className='new__note__content'
                        name='content' 
                        id='content' 
                        onChange={e => this.updateContent(e.target.value)}
                    />  
                </div>
                <div className='NoteListNav__button-wrapper'>
                    <CircleButton
                        className='NoteListNav__add-note-button'
                        onClick={e=> this.handleSubmit(e)}
                    >
                        <FontAwesomeIcon icon='plus' />
                        <br />
                        Submit
                    </CircleButton>
                    </div>
            </form>
            </div>
        )
    }

}

export default AddNote