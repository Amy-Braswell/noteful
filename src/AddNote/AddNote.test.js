import React from 'react'
import renderer from 'react-test-renderer'
import AddNote from './AddNote'

describe(`AddNote component`, () => {
  const stubFolders = [
    {
      "id": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
      "name": "Important"
    },
    {
      "id": "b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1",
      "name": "Super"
    },
    {
      "id": "b07162f0-ffaf-11e8-8eb2-f2801f1b9fd1",
      "name": "Spangley"
    }
  ]

  it('renders the complete form', () => {
    const tree = renderer
    .create(<AddNote />)
    .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders the select options from folders', () => {
    const select = renderer
    .create(<AddNote folders={stubFolders} />)
    .find('#note-folder-select')
    .toJSON()
    expect(select).toMatchSnapshot()
  })
})