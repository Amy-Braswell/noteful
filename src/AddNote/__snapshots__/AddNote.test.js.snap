// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNote component renders the complete form 1`] = `
<section
  className="AddNote"
>
  <h3>
    Create a note
  </h3>
  <form
    action="#"
    className="Noteful-form AddNote__Form"
    onSubmit={[Function]}
  >
    <div>
      <p
        className="error__message"
      >
        
      </p>
    </div>
    <div>
      <p
        className="error__message"
      >
        
      </p>
    </div>
    <div
      className="field form-group"
    >
      <label
        htmlFor="note-name"
      >
        <input
          aria-required="true"
          className="new__note__name"
          defaultValue="Enter Name"
          id="note-name-input"
          name="note"
          onChange={[Function]}
          type="text"
        />
      </label>
    </div>
    <div
      className="field form-group"
    >
      <label
        htmlFor="note-content"
      >
        <textarea
          className="new__note__content"
          id="note-content-input"
          name="content"
          onChange={[Function]}
        />
      </label>
    </div>
    <div
      className="field form-group"
    >
      <label
        htmlFor="select-folder"
      >
        <select
          aria-required="true"
          id="note-folder-select"
          name="note-folder-id"
          onChange={[Function]}
        >
          <option
            value={null}
          >
            Select Folder
          </option>
        </select>
      </label>
    </div>
    <div
      className="buttons"
    >
      <button
        type="submit"
      >
        <br />
        Note
      </button>
    </div>
  </form>
</section>
`;
